File: src/main/java/simulation/Vehicles/Car.java
package simulation.Vehicles;

/*
 *   @version: 1.0
 *   @since: 04.04.2020
 */
public class Car extends Vehicle{
    /**
     * Car class  for Nagel-Schreckenberg model extending Vehicle abtract class.
     *
     */

    public Car(int x, int y) {
        super(x, y);
    }

    public Car(int x, int y, int velocity) {
        super(x, y, velocity);
    }

    public Car(int x, int y, int velocity, int maxVelocity, int acceleration, double brakeProbability){
        super(x, y, velocity, maxVelocity, acceleration, brakeProbability);
    }

    @Override
    public String toString() {
        return "<Car pos=(" + this.x + ", " + this.y + ") dist="+ this.distanceToForwardingVehicle + ">";
    }
}

File: src/objects/Car.js
import Vehicle from './Vehicle';
import { VEHICLE_TYPE } from '../consts/vehicles.const';

export default class Car extends Vehicle {
  /**
   * Car Object extending base Vehicle for Nagel-Schreckenberg model.
   *
   * @param {number} id
   * @param {number} x
   * @param {number} y
   * @param {number} destinationId
   * @param {number} velocity
   * @param {number} maxVelocity
   */
  constructor(id, x, y, destinationId, velocity, maxVelocity) {
    super(id, x, y, destinationId, VEHICLE_TYPE.CAR, velocity, maxVelocity);
  }
}
